server/middleware/OktaServerSideAuthMiddleware.ts

```diff
@@ -0,0 +1,90 @@
+import { joinUrl } from '@guardian/libs';
+import type { NextFunction, Request, Response } from 'express';
+import fetch from 'node-fetch';
+import {
+	clearIdentityLocalState,
+	getIdentityLocalState,
+	setIdentityLocalState,
+} from '@/server/IdentityLocalState';
+import { OAuthAccessTokenCookieName } from '@/server/oauth';
+import { getConfig as getOktaConfig } from '@/server/oktaConfig';
+import { requiresSignin } from '../../shared/requiresSignin';
+
+declare const CYPRESS: string;
+
+type OktaUserInfo = {
+	legacy_identity_id: string;
+	email: string;
+	name: string;
+};
+
+export const withOktaSeverSideValidation = async (
+	req: Request,
+	res: Response,
+	next: NextFunction,
+) => {
+	if (CYPRESS === 'SKIP_IDAPI') {
+		return next();
+	}
+	const oktaConfig = await getOktaConfig();
+	if (!oktaConfig.useOkta) {
+		console.log('Okta disabled, skipping Okta server side validation...');
+		return next();
+	}
+
+	console.log('Validating token server side ...');
+	const signinRequired = () => requiresSignin(req.originalUrl);
```

Comment:
I can do that and it probably won't be much of a difference, but I would be executing it always instead of just having a function that I can call if I need it, right ?

Reply:
Well, it's a function which checks if a string begins with something so I'd say the performance hit is negligible. It just feels cleaner to me this way, doesn't make much of a difference.